tracer/src/Datadog.Trace/Ci/CITracerManagerFactory.cs

```diff
@@ -34,12 +46,78 @@ internal class CITracerManagerFactory : TracerManagerFactory
 
         protected override ISampler GetSampler(ImmutableTracerSettings settings)
         {
+            if (!string.IsNullOrEmpty(_settings.ApiKey))
```

Comment:
Why do we return null here? Are there going to be any issues with returning a null sampler?

Reply:
Technically is not required to sample anything in an Agentless scenario, because the SamplingPriority set by the sampler is not being used. That's the reason to return null here. Although it's true that I'm not 100% sure this would no cause issue with other tracer components. So, to play safe, I'll return the CISampler instance (that always set the value to UserKeep).

Reply:
done in [c669f09](https://github.com/DataDog/dd-trace-dotnet/pull/2402/commits/c669f0981cfbcd30ba37607a8686b90d11cc7f26)